src/shims/windows/foreign_items.rs

```diff
@@ -340,14 +389,48 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 // std-only shim.
                 this.write_scalar(Scalar::from_machine_isize(which.into(), this), dest)?;
             }
+            "CloseHandle" => {
+                let [handle] =
+                    this.check_shim(abi, Abi::System { unwind: false }, link_name, args)?;
+
+                match Handle::from_scalar(this.read_scalar(handle)?.check_init()?, this)? {
+                    Some(Handle::Thread(thread)) => this.detach_thread(thread)?,
+                    _ => throw_ub_format!("invalid handle"),
+                };
+
+                this.write_scalar(Scalar::from_u32(1), dest)?;
+            }
 
-            // Better error for attempts to create a thread
+            // Threading
             "CreateThread" => {
-                let [_, _, _, _, _, _] =
+                let [_security, _stacksize, start, arg, _flags, thread] =
                     this.check_shim(abi, Abi::System { unwind: false }, link_name, args)?;
```

Comment:
For shims that we actually support, each argument needs to be checked. The least this should do is to call `to_X` for the right `X` to ensure that all arguments have the right size.